src/Microsoft.Health.Api/Features/ByteCounter/ByteCountingStream.cs

```diff
@@ -0,0 +1,135 @@
+        /// The number of bytes that have been written.
+        /// </summary>
+        public long WrittenByteCount { get => _writtenByteCount; }
+
+        /// <inheritdoc />
+        public override void Flush()
+        {
+            _stream.Flush();
+        }
+
+        /// <inheritdoc />
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            return _stream.FlushAsync(cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return _stream.Read(buffer, offset, count);
+        }
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            return _stream.Seek(offset, origin);
+        }
+
+        /// <inheritdoc />
+        public override void SetLength(long value)
+        {
+            _stream.SetLength(value);
+        }
+
+        /// <inheritdoc />
+        public override void WriteByte(byte value)
+        {
+            _stream.WriteByte(value);
+            Interlocked.Increment(ref _writtenByteCount);
+        }
+
+        /// <inheritdoc />
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            _stream.Write(buffer, offset, count);
+            Interlocked.Add(ref _writtenByteCount, count);
+        }
+
+        /// <summary>
+        /// Asynchronously writes a sequence of bytes to the current stream and advances the current position within this stream by the number of bytes written.
+        /// </summary>
+        /// <param name="buffer">The buffer to write data from.</param>
+        /// <param name="offset">The zero-based byte offset in buffer from which to begin copying bytes to the stream.</param>
+        /// <param name="count">The maximum number of bytes to write.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public new Task WriteAsync(byte[] buffer, int offset, int count)
+        {
+            Interlocked.Add(ref _writtenByteCount, count);
+            return _stream.WriteAsync(buffer, offset, count);
+        }
+
+        /// <inheritdoc />
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            Interlocked.Add(ref _writtenByteCount, count);
+            return _stream.WriteAsync(buffer, offset, count, cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, object state)
+        {
+            Interlocked.Add(ref _writtenByteCount, count);
+            return _stream.BeginWrite(buffer, offset, count, callback, state);
+        }
+
+        /// <inheritdoc />
+        protected override void Dispose(bool disposing)
+        {
+            _stream.Dispose();
+            base.Dispose(disposing);
```

Comment:
If we seal this class, we could probably simplify this to simply do something like:
```csharp
public void Dispose()
  => _stream.Dipose();
```

Reply:
Even in a sealed class, removing "base.Dispose(disposing);" will pop up an error complaining that base.Dispose(disposing) is missing. I don't think it's worth it to suppress the error so I left it as is